File: python/3.TimeComplexity/TapeEquilibrium.py
def solution(A):
    length = len(A)
    right_side = sum(A)
    min_diff = float('inf')
    left_side = 0
    for i in range(0, length - 1):
        left_side += A[i]
        right_side -= A[i]
        diff = abs(left_side - right_side)
        min_diff = min(min_diff, diff)
    return min_diff

File: javascript/3.TimeComplexity/TapeEquilibrium.js
function solution(A) {
  const length = A.length;
  let rightSide = A.reduce((sum, num) => sum + num, 0);
  let minDiff = Infinity;
  let leftSide = 0;

  for (let i = 0; i < length - 1; i++) {
    leftSide += A[i];
    rightSide -= A[i];
    const diff = Math.abs(leftSide - rightSide);
    minDiff = Math.min(minDiff, diff);
  }

  return minDiff;
}
